CONTRIBUTING.md

```diff
@@ -1,15 +1,158 @@
+
+```
+<type>(<scope>): <subject>
+<BLANK LINE>
+<body>
+<BLANK LINE>
+<footer>
+```
+The **header** is mandatory and the **scope** of the header is optional.
+
+Any line of the commit message cannot be longer 100 characters! This allows the message to be easier
+to read on GitHub as well as in various git tools.
+
+#### Type
+Must be one of the following:
+
+- **feat**: A new feature
+- **fix**: A bug fix
+- **docs**: Documentation only changes
+- **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing
+             semi-colons, etc)
+- **refactor**: A code change that neither fixes a bug nor adds a feature
+- **perf**: A code change that improves performance
+- **test**: Adding missing tests
+- **chore**: Changes to the build process or auxiliary tools and libraries such as documentation
+generation
+- **revert**: reverts a previous commit. The subject should be the header of the reverted commit. In the body it should say: `This reverts commit <hash>.`, where the hash is the SHA of the commit being reverted.
+
+#### Scope
+The scope could be anything specifying place of the commit change. For example `$location`,
+`$browser`, `$compile`, `$rootScope`, `ngHref`, `ngClick`, `ngView`, etc...
+
+#### Subject
+The subject contains succinct description of the change:
+
+* use the imperative, present tense: "change" not "changed" nor "changes"
+* do not capitalize first letter
+* no dot (.) at the end
+
+#### Body
+Just as in the **subject**, use the imperative, present tense: "change" not "changed" nor "changes".
+The body should include the motivation for the change and contrast this with previous behavior.
+
+#### Footer
+The footer should contain any information about **Breaking Changes** and is also the place to
+reference GitHub issues that this commit **Closes**.
+
+**Breaking Changes** should start with the word `BREAKING CHANGE:` with a space or two newlines. The rest of the commit message is then used for this.
+
+### Submit a Pull Request
+
+
+1. Make your changes in a new git branch:
+
+```shell
+git checkout -b my-pr-branch master
+```
+2. Create your patch, **including appropriate test cases**.
+3. Commit your changes using a descriptive commit message that follows our
+    [commit message format](#Commit_Message_Format).
+
+```shell
+git commit -a
+```
+Note: the optional commit `-a` command line option will automatically "add" and "rm" edited files.
+
+
+4. Push your branch to GitHub:
+
+```shell
+git push --set-upstream origin my-pr-branch
+```
+
+5. In GitHub, target the pull request to `zanata:master`.
+6. If we suggest changes then:
+   a. Make the required updates.
+   b. Commit your changes to your branch (e.g. `my-pr-branch`).
+    * Push the changes to your GitHub repository (this will update your Pull Request).
+
+7. If the PR gets too outdated we may ask you to rebase and force push to
+update the pull request:
+```shell
+git rebase master -i
+git push origin my-pr-branch -f
```

Comment:
Force push is dangerous (someone could lose all their work), so we should probably not include this instruction here. We can help people who are not familiar with git if we ask them to rebase and they respond that they don't know how.